### ISSUE TITLE
StrawberryShake generator gets confused over raw json in schema

### ISSUE THREAD

**Issue by niles11** [ChilliCream/graphql-platform#6257]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Strawberry Shake

### Describe the bug

Hi, i'm using StrawberryShake to connect to a [ContentStack GraphQL interface](https://www.contentstack.com/docs/developers/apis/graphql-content-delivery-api/). ContentStack is a CMS, which means you can add content models in the CMS and after doing that the schema will automatically update accordingly. The StrawberryShake experience so far has been good, even with man, many different content models which all have nested fields.

However, recently I started using a field type that we had not used before in any content model: the [JSON RTE field](https://www.contentstack.com/docs/developers/json-rich-text-editor/). This field accepts markdown text but it will be converted into json. After running the command `dotnet graphql update ...` and doing a rebuild of the solution I get the following error:

```An item with the same key has already been added. Key: JSON```

I do see these changes in the schema.graphql file:

<img width="499" alt="2023-06-12 11_32_02-Sourcetree" src="https://github.com/ChilliCream/graphql-platform/assets/7943393/d34f987b-6ae9-4f6c-8c05-cf8b05b6afaa">

The changes seem to suggest there is support for 'raw' json within the graphql response. But the solution no longer compiles because of the error mentioned above. It should not be related to [this](https://stackoverflow.com/questions/75975790/exec-error-gql-an-item-with-the-same-key-has-already-been-added-key-json) StackOverflow issue, because without the JSON RTE field in the content model StrawberryShake immediately works perfectly fine.

### Steps to reproduce

1. Create a (in my case .net 6) project
2. Setup StrawberryShake
3. Connect to ContentStack CMS (sorry, you need an account for this, there is no ContentStack playground)
4. Add model without JSON RTE field
5. Update the schema.json with update command
6. Notice it works
7. Add JSON RTE field to the model which outputs 'raw' JSON
8. Update the schema.json again with update command
9. The solution no longer builds and throws the error defined below in the next question

### Relevant log output

```shell
An item with the same key has already been added. Key: JSON
```

### Version

13.2.0

**Comment by ademchenko** [comment]
This is a very serious issue which is easily reproduced on any schema having the definition of scalar JSON.

I have managed to make the reproducing example even for the Hotchocolate server.

Suppose we have the very basic schema:
```
schema {
  query: Query
}

type Query {
  persons: [Person!]!
}

type Person {
  data: JSON
}

scalar JSON
```
Or in terms of code first:
```

namespace Demo.Server;

public class Person
{
    public System.Text.Json.JsonElement? Data { get; set; }
}

public class Query
{
    public ICollection<Person> GetPersons() => Array.Empty<Person>();
}

public class QueryType : ObjectType<Query> { }
```
Then any attempt of generating the client (whether with _dotnet build_ or with _dotnet graphql generate_ ) causes the error _"EXEC : error GQL: An item with the same key has already been added. Key: JSON"_

**Comment by chenglidev** [comment]
I encountered the same issue.
Is there any patch scheduled?

**Comment by nightroman** [comment]
Same issue. After reading the encouraging What's New in v13 about JSON - https://chillicream.com/blog/2023/02/08/new-in-hot-chocolate-13#json-scalar tried to engage in our project. Alas...

**Comment by sfnmk** [comment]
Any news on this? I cannot use the client as it is right now.

**Comment by niles11** [comment]
It seems this is fixed since [v13.8.0](https://github.com/ChilliCream/graphql-platform/releases/tag/13.8.0), which has been released yesterday. This exact issue is not resolved or mentioned in the releasenotes so I only noticed the change by coincidence. The releasenotes do mention these two fixes which solve my issue as well 🥳:
- Fix JsonElement serialization for StrawberryShake
- Reworked JSON scalar usage with StrawberryShake. (https://github.com/ChilliCream/graphql-platform/pull/6683)
